site/content/protocols/messagepickup/4.0/readme.md

```diff
@@ -0,0 +1,388 @@
+
+The `messages-received` message is sent by the `recipient` to confirm receipt of delivered messages, prompting the `mediator` to clear messages from the queue.
+
+The `live-delivery-change` message is used to set the state of `live_delivery`.
+
+When _Live Mode_ is enabled, messages that arrive when an existing connection exists are delivered over the connection immediately, via a `delivery` message, rather than being pushed to the queue. See _Live Mode_ below for more details. 
+
+## Security
+
+This protocol expects messages to be encrypted during transmission, and repudiable. 
+
+## Message Reference
+
+### Status Request
+Sent by the `recipient` to the `mediator` to request a `status` message.
+
+Message Type URI: `https://didcomm.org/message-pickup/4.0/status-request`
+
+DIDComm v1 example:
+```json
+{
+    "@id": "123456780",
+    "@type": "https://didcomm.org/message-pickup/4.0/status-request",
+    "recipient_did": "<did:key for messages>",
+    "~transport": {
+        "return_route": "all"
+    }
+}
+```
+
+DIDComm v2 example:
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/message-pickup/4.0/status-request",
+    "body" : {
+        "recipient_did": "<did for messages>"
+    },
+    "return_route": "all"
+}
+```
+`recipient_did` is optional. When specified, the `mediator` **MUST** only return status related to that recipient did. This allows the `recipient` to discover if any messages are in the queue that were sent to a specific DID. If using DIDComm v1, `recipient_did` **MUST** be a [`did:key` reference](https://github.com/hyperledger/aries-rfcs/tree/main/features/0360-use-did-key).
+
+### Status
+Status details about waiting messages.
+
+Message Type URI: `https://didcomm.org/message-pickup/4.0/status`
+
+DIDComm v1 example:
+```json
+{
+    "@id": "123456780",
+    "@type": "https://didcomm.org/message-pickup/4.0/status",
+    "~thread": {
+        "thid": "<message id of status-request message>"
+    },
+    "recipient_did": "<did:key for messages>",
+    "message_count": 7,
+    "longest_waited_seconds": 3600,
+    "newest_received_time": 1658085169,
+    "oldest_received_time": 1658084293,
+    "total_bytes": 8096,
+    "live_delivery": false
+}
+```
+
+DIDComm v2 example:
+```json
+{
+    "id": "123456780",
+    "thid": "<message id of status-request message>",
+    "type": "https://didcomm.org/message-pickup/4.0/status",
+    "body": {
+        "recipient_did": "<did for messages>",
+        "message_count": 7,
+        "longest_waited_seconds": 3600,
+        "newest_received_time": 1658085169,
+        "oldest_received_time": 1658084293,
+        "total_bytes": 8096,
+        "live_delivery": false
+    }
+}
+```
+
+`message_count` is the only **REQUIRED** attribute. The others **MAY** be present if offered by the `mediator`.
+
+`longest_waited_seconds` is in seconds, and is the longest delay of any message in the queue.
```

Comment:
I _think_ this is how long the oldest message has been waiting for delivery, but I do think it could use some clarification. @dbluhm do you know if this is supposed to mean something different than I described (since you've done significant amounts of implementation of these protocols)?

Reply:
Honestly speaking, I don't think I've ever implemented this portion. But yeah, I think it is the "age" in seconds of the oldest message on the queue.